core/src/avm2/globals/flash/geom/transform.rs

```diff
@@ -45,6 +45,32 @@ pub fn set_color_transform<'gc>(
     Ok(Value::Undefined)
 }
 
+pub fn get_matrix_3d<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    // FIXME: This Matrix3D is generated from the 2D Matrix.
+    // It does not work when the 2D matrix contains any transformation in 3D.
+    // Support native Matrix3D.
```

Comment:
We should preserve the stub warnings using `avm2_stub_getter!`, `avm2_stub_setter!`, as the 3D matrix is not yet fully supported. In order to do that, add the following under this comment (+ the other one in the setter).

```rust
avm2_stub_getter!(activation, "flash.geom.Transform", "matrix3D");
```

Reply:
Thank you, true. I didn't know that we can produce the warning from Rust, too! Fixed: [62d6f79](https://github.com/ruffle-rs/ruffle/pull/18810/commits/62d6f794ee02b0f22a9e5d41d5d1917e9bb77c8c)